### ISSUE TITLE
Calling `dict()` on an `Any` type gets inferred as `dict[str, str]` instead of `dict[Any, Any]`

### ISSUE THREAD

**Issue by Pyprohly** [microsoft/pyright#4490]
```python
from typing import Any
a: Any = ...
d = dict(a)
reveal_type(d)  # Type of "d" is "dict[str, str]"
```

Regression? This used to get inferred as `dict[Any, Any]`.

Bisecting reveals 1dc84b9 (#4347) (included in v1.1.285) is the bad commit.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.291, which I just published. It will also be included in a future release of pylance.
